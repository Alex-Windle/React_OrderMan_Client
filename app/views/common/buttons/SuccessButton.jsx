/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 */


// Module imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';


/**
 * SuccessButton is a button used for success-type actions.
 *
 * @description Renders a Bootstrap button
 *
 * PROPS:
 *   block {bool} - If button is a 'btn-block'
 *   disabled {bool} - If the button is disabled
 *   loading {bool} - If the button is loading
 *   title {string} - Button title
 *   onClick {func} - Button onClick handler
 *
 * @extends Component
 */
export default class SuccessButton extends Component {
  render() {
    const {
      block,
      disabled,
      loading,
      title,
      onClick,
    } = this.props;

    return (
      <button
        type="submit"
        className={`btn btn-info ${block ? 'btn-block' : ''}`}
        disabled={disabled || loading}
        onClick={onClick}
      >
        {(() => (loading ? <i className="fa fa-spin fa-spinner" /> : title))()}
      </button>
    );
  }
}


SuccessButton.propTypes = {
  block: PropTypes.bool,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};


SuccessButton.defaultProps = {
  block: false,
  disabled: false,
  loading: false,
  onClick: () => {},
};
