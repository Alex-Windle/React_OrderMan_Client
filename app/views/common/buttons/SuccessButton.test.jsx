/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 */


// Module imports
import React from 'react';
import renderer from 'react-test-renderer';


// Component imports
import SuccessButton from './SuccessButton';


describe('SuccessButton component', () => {
  it('matches snapshot- not loading, not disabled, not block', () => {
    const tree = renderer.create(
      <SuccessButton
        title="Go"
        onClick={jest.fn()}
        loading={false}
        disabled={false}
        block={false}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot- not loading, not disabled, block', () => {
    const tree = renderer.create(
      <SuccessButton
        block
        title="Go"
        onClick={jest.fn()}
        loading={false}
        disabled={false}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot- not loading, disabled, block', () => {
    const tree = renderer.create(
      <SuccessButton
        disabled
        title="Go"
        onClick={jest.fn()}
        loading={false}
        block={false}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot- loading, disabled, block', () => {
    const tree = renderer.create(
      <SuccessButton
        block
        disabled
        loading
        title="Go"
        onClick={jest.fn()}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
