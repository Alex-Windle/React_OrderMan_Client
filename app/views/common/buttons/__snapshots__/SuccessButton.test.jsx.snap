// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessButton component matches snapshot- loading, disabled, block 1`] = `
<button
  className="btn btn-info btn-block"
  disabled={true}
  onClick={[Function]}
  type="submit"
>
  <i
    className="fa fa-spin fa-spinner"
  />
</button>
`;

exports[`SuccessButton component matches snapshot- not loading, disabled, block 1`] = `
<button
  className="btn btn-info "
  disabled={true}
  onClick={[Function]}
  type="submit"
>
  Go
</button>
`;

exports[`SuccessButton component matches snapshot- not loading, not disabled, block 1`] = `
<button
  className="btn btn-info btn-block"
  disabled={false}
  onClick={[Function]}
  type="submit"
>
  Go
</button>
`;

exports[`SuccessButton component matches snapshot- not loading, not disabled, not block 1`] = `
<button
  className="btn btn-info "
  disabled={false}
  onClick={[Function]}
  type="submit"
>
  Go
</button>
`;
